TODO:

Change negative values used in minmax to 0

Make function to check what countries are not in the dataset

Make sure UK is included (name varies)

In [2]:
import numpy as np
import pandas as pd
import os
import requests
import json
from xml.etree import ElementTree as ET

In [3]:
# set parent folder as working directory
os.chdir("..")
os.getcwd()

'c:\\Users\\isaldiviagonzatti\\OneDrive - Kiel Institut für Weltwirtschaft\\Documents\\General\\Ignacio'

In [4]:
# define paths, create output folder if not there
data_path = 'data'
output_path = 'output'

if not os.path.exists(output_path):
	os.makedirs(output_path)

#### Country names dict

In [5]:
country_to_abbrev = {
    "Andorra": "AD",
    "United Arab Emirates": "AE",
    "Afghanistan": "AF",
    "Antigua and Barbuda": "AG",
    "Anguilla": "AI",
    "Albania": "AL",
    "Armenia": "AM",
    "Angola": "AO",
    "Antarctica": "AQ",
    "Argentina": "AR",
    "American Samoa": "AS",
    "Austria": "AT",
    "Australia": "AU",
    "Aruba": "AW",
    "Åland Islands": "AX",
    "Azerbaijan": "AZ",
    "Bosnia and Herzegovina": "BA",
    "Barbados": "BB",
    "Bangladesh": "BD",
    "Belgium": "BE",
    "Burkina Faso": "BF",
    "Bulgaria": "BG",
    "Bahrain": "BH",
    "Burundi": "BI",
    "Benin": "BJ",
    "Saint Barthélemy": "BL",
    "Bermuda": "BM",
    "Brunei Darussalam": "BN",
    "Bolivia (Plurinational State of)": "BO",
    "Bonaire, Sint Eustatius and Saba": "BQ",
    "Brazil": "BR",
    "Bahamas": "BS",
    "Bhutan": "BT",
    "Bouvet Island": "BV",
    "Botswana": "BW",
    "Belarus": "BY",
    "Belize": "BZ",
    "Canada": "CA",
    "Cocos (Keeling) Islands": "CC",
    "Congo, Democratic Republic of the": "CD",
    "Central African Republic": "CF",
    "Congo": "CG",
    "Switzerland": "CH",
    "Côte d'Ivoire": "CI",
    "Cook Islands": "CK",
    "Chile": "CL",
    "Cameroon": "CM",
    "China": "CN",
    "Colombia": "CO",
    "Costa Rica": "CR",
    "Cuba": "CU",
    "Cabo Verde": "CV",
    "Curaçao": "CW",
    "Christmas Island": "CX",
    "Cyprus": "CY",
    "Czechia": "CZ",
    "Germany": "DE",
    "Djibouti": "DJ",
    "Denmark": "DK",
    "Dominica": "DM",
    "Dominican Republic": "DO",
    "Algeria": "DZ",
    "Ecuador": "EC",
    "Estonia": "EE",
    "Egypt": "EG",
    "Western Sahara": "EH",
    "Eritrea": "ER",
    "Spain": "ES",
    "Ethiopia": "ET",
    "Finland": "FI",
    "Fiji": "FJ",
    "Falkland Islands (Malvinas)": "FK",
    "Micronesia (Federated States of)": "FM",
    "Faroe Islands": "FO",
    "France": "FR",
    "Gabon": "GA",
    "United Kingdom of Great Britain and Northern Ireland": "UK", #original is GB, Eurostat uses UK
    "United Kingdom of GB": "GB",
    "United Kingdom": "UK", #original is United Kingdom of GB, OECD uses United Kingdom
    "Grenada": "GD",
    "Georgia": "GE",
    "French Guiana": "GF",
    "Guernsey": "GG",
    "Ghana": "GH",
    "Gibraltar": "GI",
    "Greenland": "GL",
    "Gambia": "GM",
    "Guinea": "GN",
    "Guadeloupe": "GP",
    "Equatorial Guinea": "GQ",
    "Greece": "EL", #original ir GR, Eurostat uses EL
    "South Georgia and the South Sandwich Islands": "GS",
    "Guatemala": "GT",
    "Guam": "GU",
    "Guinea-Bissau": "GW",
    "Guyana": "GY",
    "Hong Kong": "HK",
    "Heard Island and McDonald Islands": "HM",
    "Honduras": "HN",
    "Croatia": "HR",
    "Haiti": "HT",
    "Hungary": "HU",
    "Indonesia": "ID",
    "Ireland": "IE",
    "Israel": "IL",
    "Isle of Man": "IM",
    "India": "IN",
    "British Indian Ocean Territory": "IO",
    "Iraq": "IQ",
    "Iran (Islamic Republic of)": "IR",
    "Iceland": "IS",
    "Italy": "IT",
    "Jersey": "JE",
    "Jamaica": "JM",
    "Jordan": "JO",
    "Japan": "JP",
    "Kenya": "KE",
    "Kyrgyzstan": "KG",
    "Cambodia": "KH",
    "Kiribati": "KI",
    "Comoros": "KM",
    "Saint Kitts and Nevis": "KN",
    "Korea (Democratic People's Republic of)": "KP",
    "Korea, Republic of": "KR",
    "Kuwait": "KW",
    "Cayman Islands": "KY",
    "Kazakhstan": "KZ",
    "Lao People's Democratic Republic": "LA",
    "Lebanon": "LB",
    "Saint Lucia": "LC",
    "Liechtenstein": "LI",
    "Sri Lanka": "LK",
    "Liberia": "LR",
    "Lesotho": "LS",
    "Lithuania": "LT",
    "Luxembourg": "LU",
    "Latvia": "LV",
    "Libya": "LY",
    "Morocco": "MA",
    "Monaco": "MC",
    "Moldova, Republic of": "MD",
    "Montenegro": "ME",
    "Saint Martin (French part)": "MF",
    "Madagascar": "MG",
    "Marshall Islands": "MH",
    "North Macedonia": "MK",
    "Mali": "ML",
    "Myanmar": "MM",
    "Mongolia": "MN",
    "Macao": "MO",
    "Northern Mariana Islands": "MP",
    "Martinique": "MQ",
    "Mauritania": "MR",
    "Montserrat": "MS",
    "Malta": "MT",
    "Mauritius": "MU",
    "Maldives": "MV",
    "Malawi": "MW",
    "Mexico": "MX",
    "Malaysia": "MY",
    "Mozambique": "MZ",
    "Namibia": "NA",
    "New Caledonia": "NC",
    "Niger": "NE",
    "Norfolk Island": "NF",
    "Nigeria": "NG",
    "Nicaragua": "NI",
    "Netherlands": "NL",
    "Norway": "NO",
    "Nepal": "NP",
    "Nauru": "NR",
    "Niue": "NU",
    "New Zealand": "NZ",
    "Oman": "OM",
    "Panama": "PA",
    "Peru": "PE",
    "French Polynesia": "PF",
    "Papua New Guinea": "PG",
    "Philippines": "PH",
    "Pakistan": "PK",
    "Poland": "PL",
    "Saint Pierre and Miquelon": "PM",
    "Pitcairn": "PN",
    "Puerto Rico": "PR",
    "Palestine, State of": "PS",
    "Portugal": "PT",
    "Palau": "PW",
    "Paraguay": "PY",
    "Qatar": "QA",
    "Réunion": "RE",
    "Romania": "RO",
    "Serbia": "RS",
    "Russian Federation": "RU",
    "Rwanda": "RW",
    "Saudi Arabia": "SA",
    "Solomon Islands": "SB",
    "Seychelles": "SC",
    "Sudan": "SD",
    "Sweden": "SE",
    "Singapore": "SG",
    "Saint Helena, Ascension and Tristan da Cunha": "SH",
    "Slovenia": "SI",
    "Svalbard and Jan Mayen": "SJ",
    "Slovakia": "SK",
    "Sierra Leone": "SL",
    "San Marino": "SM",
    "Senegal": "SN",
    "Somalia": "SO",
    "Suriname": "SR",
    "South Sudan": "SS",
    "Sao Tome and Principe": "ST",
    "El Salvador": "SV",
    "Sint Maarten (Dutch part)": "SX",
    "Syrian Arab Republic": "SY",
    "Eswatini": "SZ",
    "Turks and Caicos Islands": "TC",
    "Chad": "TD",
    "French Southern Territories": "TF",
    "Togo": "TG",
    "Thailand": "TH",
    "Tajikistan": "TJ",
    "Tokelau": "TK",
    "Timor-Leste": "TL",
    "Turkmenistan": "TM",
    "Tunisia": "TN",
    "Tonga": "TO",
    "Turkey": "TR",
    "Trinidad and Tobago": "TT",
    "Tuvalu": "TV",
    "Taiwan, Province of China": "TW",
    "Tanzania, United Republic of": "TZ",
    "Ukraine": "UA",
    "Uganda": "UG",
    "United States Minor Outlying Islands": "UM",
    "United States of America": "US",
    "Uruguay": "UY",
    "Uzbekistan": "UZ",
    "Holy See": "VA",
    "Saint Vincent and the Grenadines": "VC",
    "Venezuela (Bolivarian Republic of)": "VE",
    "Virgin Islands (British)": "VG",
    "Virgin Islands (U.S.)": "VI",
    "Viet Nam": "VN",
    "Vanuatu": "VU",
    "Wallis and Futuna": "WF",
    "Samoa": "WS",
    "Yemen": "YE",
    "Mayotte": "YT",
    "South Africa": "ZA",
    "Zambia": "ZM",
    "Zimbabwe": "ZW",
}
    
# invert the dictionary
abbrev_to_country = dict(map(reversed, country_to_abbrev.items()))

In [ ]:
allEurope = pd.read_csv('./data/Countries-Europe.csv')
allEurope = allEurope.name.to_list()

In [64]:
countries=['Belgium','Bulgaria','Cyprus', 'Greece','Germany','Croatia','Italy', 
           'Denmark','Estonia','Spain','Finland','France','Ireland','Lithuania',
           'Latvia','Malta','Netherlands','Poland','Portugal', 'Romania',
           'Sweden','United Kingdom of Great Britain and Northern Ireland', 'United Kingdom']
countryAbb = [x if x not in country_to_abbrev else country_to_abbrev[x] for x in countries]

### SDG Official Data

In [98]:
%%time 
# SDG14 indicators from the UNstats
# https://unstats.un.org/sdgs/dataportal/database

sdg14 = pd.read_excel(os.path.join(data_path, "Goal14_april2023.xlsx"), sheet_name=0)
sdg14 = sdg14[['Target', 'Indicator','SeriesCode', 'GeoAreaName',
                   'TimePeriod', 'Value','Units','SeriesDescription', ]]

sdg14.Value.replace('N', np.nan, inplace=True)
sdg14.Value = sdg14.Value.astype(np.float64)
                  
# filter countries
# sdg14 = sdg14[sdg14['GeoAreaName'].isin(countries)]
# show indicators
pd.DataFrame(sdg14.groupby(['Indicator', 'SeriesCode', 'SeriesDescription', 'Units']).size())

Wall time: 23.7 s


0
Indicator SeriesCode        SeriesDescription                                  Units        
14.1.1    EN_MAR_BEALITSQ   Beach litter per square kilometer (Number)         NUMBER    611
          EN_MAR_BEALIT_BP  Beach litter originating from national land-bas... PERCENT   900
          EN_MAR_BEALIT_BV  Beach litter originating from national land-bas... TONNES    900
          EN_MAR_BEALIT_EXP Exported beach litter originating from national... TONNES    900
          EN_MAR_BEALIT_OP  Beach litter originating from national land-bas... PERCENT   900
          EN_MAR_BEALIT_OV  Beach litter originating from national land-bas... TONNES    900
          EN_MAR_CHLANM     Chlorophyll-a anomaly, remote sensing (%)          PERCENT  2784
          EN_MAR_CHLDEV     Chlorophyll-a deviations, remote sensing (%)       PERCENT  4131
          EN_MAR_PLASDD     Floating plastic debris density (count per km2)    NUMBER      3
14.2.1    EN_SCP_ECSYBA     Number of countries using ecosystem-based appro... NUMBER     33
14.3.1    ER_OAW_MNACD      Average marine acidity (pH) measured at agreed ... PH        903
14.4.1    ER_H2O_FWTL       Proportion of fish stocks within biologically s... PERCENT   422
14.5.1    ER_MRN_MPA        Average proportion of Marine Key Biodiversity A... PERCENT  6049
14.6.1    ER_REG_UNFCIM     Progress by countries in the degree of implemen... NUMBER    882
14.7.1    EN_SCP_FSHGDP     Sustainable fisheries as a proportion of GDP       PERCENT  5880
14.a.1    ER_RDE_OSEX       National ocean science expenditure as a share o... PERCENT   159
14.b.1    ER_REG_SSFRAR     Degree of application of a legal/regulatory/pol... NUMBER    882
14.c.1    ER_UNCLOS_IMPLE   Score for the implementation of UNCLOS and its ... PERCENT    63
          ER_UNCLOS_RATACC  Score for the ratification of and accession to ... PERCENT    63

In [152]:
sdg14[sdg14['SeriesCode']=='ER_UNCLOS_RATACC'].pivot_table(columns='TimePeriod', index='GeoAreaName', values='Value', aggfunc='mean')

TimePeriod                                           2021   2023
GeoAreaName                                                     
Argentina                                            67.0   67.0
Australia                                             NaN  100.0
Austria                                             100.0    NaN
Azerbaijan                                           67.0    NaN
Bangladesh                                          100.0    NaN
Belarus                                              67.0    NaN
Belize                                              100.0    NaN
Brazil                                                NaN  100.0
Bulgaria                                            100.0    NaN
Burundi                                               0.0    NaN
China                                               100.0    NaN
Colombia                                              0.0    NaN
Croatia                                             100.0    NaN
Czechia                                             100.0    NaN
Denmark                                             100.0    NaN
Ecuador                                             100.0    NaN
Estonia                                             100.0    NaN
European Union                                        NaN  100.0
France                                                NaN  100.0
Germany                                             100.0    NaN
Greece                                              100.0    NaN
Guyana                                               67.0    NaN
Iceland                                             100.0  100.0
Iraq                                                 33.0    NaN
Ireland                                             100.0    NaN
Japan                                               100.0    NaN
Kazakhstan                                            0.0    NaN
Kuwait                                                NaN   67.0
Latvia                                                NaN  100.0
Liechtenstein                                         0.0    NaN
Lithuania                                           100.0    NaN
Madagascar                                           67.0    NaN
Malaysia                                             67.0    NaN
Mauritius                                           100.0    NaN
Montenegro                                           67.0    NaN
Namibia                                             100.0    NaN
New Zealand                                         100.0  100.0
Pakistan                                             67.0    NaN
Panama                                              100.0    NaN
Peru                                                  0.0    NaN
Philippines                                         100.0    NaN
Portugal                                            100.0  100.0
Qatar                                                67.0   67.0
Republic of Korea                                   100.0    NaN
Romania                                             100.0    NaN
Russian Federation                                    NaN  100.0
Saudi Arabia                                          NaN   67.0
Singapore                                             NaN   67.0
Slovakia                                            100.0    NaN
Slovenia                                            100.0  100.0
South Sudan                                           0.0    NaN
Suriname                                             67.0    NaN
Switzerland                                          67.0   67.0
Togo                                                  NaN  100.0
United Kingdom of Great Britain and Northern Ir...  100.0    NaN
Venezuela (Bolivarian Republic of)                    NaN    0.0

### 14.1
#### (a) Index of coastal eutrophication

We use Gross Nitrogen Balance: Max-Min transformation where the max-value is the maximum value in the period 2012-2021 and the min-value is zero.

In [10]:
#Gross nutrient balance (Nitrogen kg/ha), Eurostat
# https://ec.europa.eu/eurostat/databrowser/view/AEI_PR_GNB__custom_153613/

nitro = pd.read_csv(os.path.join(data_path,'aei_pr_gnb.csv'))
nitro = nitro[nitro['geo'].isin(countryAbb)]
nitro['geo'] = nitro['geo'].map(abbrev_to_country).fillna(nitro['geo'])
nitro = nitro.pivot_table(columns='TIME_PERIOD', index='geo', values='OBS_VALUE', aggfunc='mean')

mr = nitro.columns[-1] # most recent year
nitro[[2012, 2014, mr]]

#maxMin transformation as (max-x)/(max-min) * 100 --> converts to 0-100 scale with 0=worst, 100=best
((nitro.loc[:,'2012':mr].max().max()  - nitro[[2012,2014, mr]])/(nitro.loc[:,'2012':mr].max().max()
                                                                   - nitro.loc[:,'2012':mr].min().min())*100).round(2)


TIME_PERIOD      2012   2014    2018
geo                                 
Belgium         23.68  28.78     NaN
Bulgaria        76.90  75.38   60.07
Croatia         49.10  66.11   61.95
Cyprus           5.19   0.76     NaN
Denmark         50.40  52.01     NaN
Estonia         75.20    NaN     NaN
Finland         66.47  66.47   61.77
France          77.22  74.49   69.47
Germany         54.12  58.24   52.86
Greece          65.26  61.50     NaN
Ireland         72.02  67.28     NaN
Italy           52.10  58.33     NaN
Latvia          76.99  75.29   75.92
Lithuania       74.66  76.63   68.26
Malta           24.84  21.84     NaN
Netherlands     11.64  14.95    0.00
Poland          66.20  69.92   60.07
Portugal        68.08  67.64   67.37
Romania         80.44  88.05  100.00
Spain           72.69  70.10     NaN
Sweden          73.77  73.99   60.61
United Kingdom  48.57  49.64     NaN

#### (b) Floating Plastic Debris Density

We use two indicators: 

1. Plastic Waste kg/ha: Max-Min Transformation where the max-value and min-values are the maximum and minimum in the period 2012-2021.

2. Recovery Rate of Plastic Packaging: Without further transformation as score as it is provided dimensionless. 

In [11]:
# Plastic Waste kg/ha, Eurostat
# https://ec.europa.eu/eurostat/databrowser/view/ENV_WASGEN/

wasteG = pd.read_csv(os.path.join(data_path,'env_wasgen.csv')) #read file
wasteG = wasteG[wasteG['geo'].isin(countryAbb)] #filter countries
wasteG['geo'] = wasteG['geo'].map(abbrev_to_country).fillna(wasteG['geo']) #change abb to name
wasteG  = wasteG[wasteG['unit']=='KG_HAB'].pivot_table(columns='TIME_PERIOD',
                                 index='geo', values='OBS_VALUE', aggfunc='mean') #pivot table 

mr = wasteG.columns[-1] # most recent year

#maxMin transformation as (max-x)/(max-min) * 100 --> converts to 0-100 scale with 0=worst, 100=best
((wasteG.loc[:,'2012':mr].max().max() - wasteG[[2012,2014, mr]])/(wasteG.loc[:,'2012':mr].max().max()
                                                                   - wasteG.loc[:,'2012':mr].min().min())*100).round(2)


TIME_PERIOD       2012   2014   2020
geo                                 
Belgium          54.62   0.00  31.93
Bulgaria         92.44  81.51  59.66
Croatia          96.64  94.12  78.15
Cyprus          100.00  99.16  88.24
Denmark          88.24  89.08  84.03
Estonia          89.92  83.19  77.31
Finland          89.92  86.55  84.03
France           83.19  82.35  73.95
Germany          78.15  76.47  73.11
Greece           94.12  89.08  90.76
Ireland          82.35  71.43  70.59
Italy            64.71  59.66  35.29
Latvia           94.96  94.12  54.62
Lithuania        89.92  83.19  70.59
Malta            95.80  93.28  89.92
Netherlands      75.63  75.63  73.11
Poland           83.19  78.15  54.62
Portugal         89.92  85.71  68.07
Romania          84.03  89.92  90.76
Spain            84.03  84.03  88.24
Sweden           89.08  78.15  76.47
United Kingdom   73.11  69.75    NaN

In [12]:
# Recovery rates for packaging waste, Plastic Packaging, Percentage, Eurostat
# https://ec.europa.eu/eurostat/databrowser/view/ten00062/

wasteR = pd.read_csv(os.path.join(data_path,'ten00062.csv'))
wasteR = wasteR[wasteR['geo'].isin(countryAbb)]
wasteR['geo'] = wasteR['geo'].map(abbrev_to_country).fillna(wasteR['geo'])
wasteR = wasteR.pivot_table(columns='TIME_PERIOD', index='geo', values='OBS_VALUE', aggfunc='mean')

mr = wasteR.columns[-1] # most recent year
wasteR = wasteR.ffill(axis = 1) #fill empty values with last available year
wasteR[[2012, 2014, mr]]

TIME_PERIOD     2012  2014   2020
geo                              
Belgium         92.7  98.3   99.4
Bulgaria        44.1  64.1   50.6
Croatia         45.4  38.0   34.1
Cyprus          44.8  46.6   51.9
Denmark         99.4  98.2   73.1
Estonia         44.0  75.9   87.4
Finland         51.0  68.3   99.4
France          64.0  64.0   71.8
Germany         99.7  99.8   99.9
Greece          32.2  32.8   37.6
Ireland         74.3  83.2  100.0
Italy           71.8  78.9   88.4
Latvia          38.7  39.0   47.9
Lithuania       38.9  51.2   63.4
Malta           33.3  33.6   10.2
Netherlands     98.1  97.9   94.9
Poland          26.2  35.8   45.0
Portugal        39.4  48.5   56.9
Romania         51.9  46.1   37.0
Spain           53.2  59.0   55.5
Sweden          58.1  56.8   50.7
United Kingdom  38.1  48.1   56.2

### 14.2

#### Proportion of national exclusive economic zones managed using ecosystem-based approaches

As of now, it is a binary variable: Number of countries using ecosystem-based approaches to manage marine areas 

Data is scarce and the next round of data collection is in 2025. More information [here](https://unstats.un.org/sdgs/metadata/files/Metadata-14-02-01.pdf)

Spatial comparison within the EU is not expected to yield great differences and we neglect target 14.2.

### 14.3

####  Average marine acidity (pH) measured at agreed suite of representative sampling stations

We use two indicators:

1. Greenhouse gas emissions under the Effort Sharing Decision (ESD): Max-Min transformation where the max- and min-values are the maximum and 
minimum in the assessment period (2012-2021)

2. Carbon Emissions per capita: Max-Min Transformation where the max-value and min-values are the maximum and minimum in the period 2012-2021.

In [13]:
# Greenhouse gas emissions under the Effort Sharing Decision (ESD), Million tonnes CO2 equivalent (Mt CO2 eq), European Environment Agency
# https://www.eea.europa.eu/data-and-maps/data/esd-4

ghgESD = pd.read_excel(os.path.join(data_path,'EEA_ESD-GHG_2022.xlsx'), sheet_name=1, skiprows=11)
ghgESD = ghgESD[ghgESD['Geographic entity'].isin(countries)]
ghgESD = ghgESD.set_index('Geographic entity')
ghgESD = ghgESD.dropna(axis=1)

mr = ghgESD.columns[-1] # most recent year
ghgESD[[2005,2012, 2016, mr]]

2005        2012        2016        2020
Geographic entity                                                
Belgium             78.942259   73.941551   74.063149   64.904157
Bulgaria            24.153081   23.239884   25.587947   25.735613
Croatia             17.199539   16.280625   16.006813   16.518244
Cyprus               4.137555    4.250998    4.111441    4.243155
Denmark             40.397354   35.677561   33.124678   30.835287
Estonia              6.283635    6.475652    6.218046    5.934829
Finland             34.008986   31.976059   31.358144   28.120096
France             392.260079  365.369122  351.924668  307.767715
Germany            469.299020  438.770433  454.157411  407.410808
Greece              62.250127   48.344795   44.897200   42.893855
Ireland             47.405961   41.511968   43.798177   44.721180
Italy              340.505320  297.128969  270.685435  254.001457
Latvia               8.069584    8.081339    9.107440    8.436245
Lithuania           11.180285   12.710486   13.921700   14.042972
Malta                1.007421    1.077945    1.329995    1.311233
Netherlands        121.319317  115.471985  101.333437   90.196821
Poland             183.803415  198.705035  198.664758  205.093211
Portugal            47.004599   39.237743   41.572594   38.536930
Romania             74.914221   72.798258   73.123042   77.123535
Spain              238.134764  204.196898  198.472205  184.188341
Sweden              42.724854   37.289991   32.612247   29.384094
United Kingdom     409.900384  343.556379  333.899779  298.936220

In [14]:
# Member State greenhouse gas emission limits in 2020 compared to 2005 greenhouse gas emissions levels
# There targets for 2020 and for 2030
# https://www.eea.europa.eu/data-and-maps/figures/national-progress-towards-greenhouse-gas
# Official journals with the data can be found at (https://climate.ec.europa.eu/eu-action/effort-sharing-member-states-emission-targets_en)

limitESR = pd.read_excel('./data/targetsESR/FIG2-154307-CLIM058-v2-Data.xlsx', sheet_name=1, skiprows=19, header=1, skipfooter=32)
limitESR = limitESR.rename(columns={'Unnamed: 0':'Country', '(Resulting) ESR target 2030 (AR4)':'2030ESRtarget','ESR limit for 2020':'2020ESRtarget', '2005 ESD BJ':'2005Level'})
limitESR = limitESR[['Country', '2020ESRtarget','2030ESRtarget','2005Level']]
limitESR.round(2)

#UK is not in the dataset, we need to add from the official journal

Country  2020ESRtarget  2030ESRtarget  2005Level
0       Austria          -0.16          -0.36      56.85
1       Belgium          -0.15          -0.35      80.29
2      Bulgaria           0.20           0.00      22.12
3       Croatia           0.11          -0.07      17.40
4        Cyprus          -0.05          -0.24       4.18
5       Czechia           0.09          -0.14      61.66
6       Denmark          -0.20          -0.39      40.08
7       Estonia           0.11          -0.13       5.43
8       Finland          -0.16          -0.39      33.94
9        France          -0.14          -0.37     398.23
10      Germany          -0.14          -0.38     477.80
11       Greece          -0.04          -0.16      62.55
12      Hungary           0.10          -0.07      48.03
13      Ireland          -0.20          -0.30      47.06
14        Italy          -0.13          -0.33     334.49
15       Latvia           0.17          -0.06       8.54
16    Lithuania           0.15          -0.09      13.25
17   Luxembourg          -0.20          -0.40      10.15
18        Malta           0.05          -0.19       1.12
19  Netherlands          -0.16          -0.36     127.81
20       Poland           0.14          -0.07     179.98
21     Portugal           0.01          -0.17      48.59
22      Romania           0.19          -0.02      75.47
23     Slovakia           0.13          -0.12      22.96
24     Slovenia           0.04          -0.14      11.83
25        Spain          -0.10          -0.26     235.99
26       Sweden          -0.17          -0.40      43.47
27          NaN            NaN            NaN        NaN
28      Iceland            NaN          -0.29        NaN
29       Norway            NaN          -0.40        NaN

In [15]:
# Population on 1 January, Eurostat
# https://ec.europa.eu/eurostat/databrowser/view/tps00001/

pop = pd.read_csv(os.path.join(data_path,'tps00001.csv'))
pop = pop[pop['geo'].isin(countryAbb)]
pop['geo'] = pop['geo'].map(abbrev_to_country).fillna(pop['geo'])
pop = pop.pivot_table(columns='TIME_PERIOD', index='geo', values='OBS_VALUE', aggfunc='mean')


In [16]:
# CO2, TOTX4_MEMONIA, THS_T thousand tonnes, Eurostat
# https://ec.europa.eu/eurostat/databrowser/view/ENV_AIR_GGE/

co2 = pd.read_csv(os.path.join(data_path,'env_air_gge.csv'))
co2 = co2[co2['geo'].isin(countryAbb)]
co2['geo'] = co2['geo'].map(abbrev_to_country).fillna(co2['geo'])
co2 = co2.pivot_table(columns='TIME_PERIOD', index='geo', values='OBS_VALUE', aggfunc='mean')

mr = co2.columns[-1] # most recent year
#co2[[2012, 2016, mr]] /1000  # million metric tonnes

co2pc = co2[[2012, 2016, 2020]]/pop[[2012, 2016, 2020]]*1000  ## tonnes co2 per capita 
co2pc

TIME_PERIOD          2012       2016      2020
geo                                           
Belgium          9.610486   9.192649  8.155780
Bulgaria         6.663881   6.447040  5.377837
Croatia          4.541044   4.410935  4.197550
Cyprus           9.398094   9.735017  8.561990
Denmark          7.658618   7.034621  5.065501
Estonia         13.563281  13.448315  7.084492
Finland          9.830334   8.967016  6.973570
France           5.875707   5.432116  4.407393
Germany         10.441041  10.061325  7.851746
Greece           8.458969   6.903288  5.311701
Ireland          8.708375   9.087713  7.318239
Italy            6.947134   6.073877  5.131773
Latvia           3.860529   3.860150  3.766501
Lithuania        4.840715   4.721411  4.944615
Malta            7.244040   3.882309  3.493735
Netherlands     10.481150  10.423888  8.323972
Poland           8.623614   8.596532  8.031805
Portugal         5.018060   5.218463  4.225050
Romania          4.510243   3.941247  3.842944
Spain            6.251912   5.973530  4.642961
Sweden           5.176026   4.663870  3.625572
United Kingdom   8.145732   6.585252  5.052171

In [17]:
# Data to corrobarate the ENV_AIR_GGE data. Very close numbers
# Fossil CO2 emissions by country (territorial), million tonnes of carbon per year (1MtC = 1 million tonne of carbon = 3.664 million tonnes of CO2)
# https://globalcarbonbudget.org/carbonbudget/

co2T = pd.read_excel(os.path.join(data_path,'National_Fossil_Carbon_Emissions_2022v1.0.xlsx'), sheet_name=1, skiprows=11) 
co2T = co2T.T
co2T.columns = co2T.iloc[0,:]
co2T.columns = co2T.columns.astype(int)
co2T = co2T.rename_axis(index='geo', columns=None)
co2T = co2T[co2T.index.isin(countries)]

mr = co2T.columns[-1] # most recent year
co2T = co2T[[2012, 2016, mr]] * 3.664 # convert from carbon to co2 

co2pc = co2T/pop[[2012, 2016, 2021]]*1000_000  ## tonnes co2 per capita 
co2pc

2012       2016      2021
geo                                           
Belgium          9.241302   8.805640  8.284225
Bulgaria         6.597165   6.358077  6.153389
Croatia          4.463862   4.321271  4.385504
Cyprus           8.426844   8.694153  8.482547
Denmark          7.144668   6.488809  5.064508
Estonia         13.427859  13.343541  7.856005
Finland          9.469202   8.598362  6.795037
France           5.467664   5.018725  4.522298
Germany         10.129644   9.739375  8.114405
Greece           8.243662   6.617751  5.273080
Ireland          8.329651   8.541494  7.498582
Italy            6.796907   5.904216  5.548756
Latvia           3.677309   3.662002  3.838013
Lithuania        4.777364   4.622091  4.965102
Malta            6.529117   3.011340  3.340736
Netherlands      9.843715   9.709692  8.071078
Poland           8.583878   8.543721  8.683397
Portugal         4.738932   4.877732  3.961567
Romania          4.490409   3.897229  4.131250
Spain            5.967441   5.625045  4.929451
Sweden           4.947727   4.407259  3.453891
United Kingdom   7.677393   6.109452       NaN

In [18]:
# These data SHALL NOT BE USED. See reason on why ENV_AIR_GGE is preferable for the calculation:
# (https://ec.europa.eu/eurostat/statistics-explained/index.php?title=Greenhouse_gas_emission_statistics_-_emission_inventories)
# (https://ec.europa.eu/eurostat/statistics-explained/index.php?title=Greenhouse_gas_emission_statistics_-_air_emissions_accounts&oldid=551152#Greenhouse_gas_emissions)
# CO2, KG_HAB, Eurostat
# https://ec.europa.eu/eurostat/databrowser/view/ENV_AC_AINAH_R2/


# co2 = pd.read_csv(os.path.join(data_path,'env_ac_ainah_r2.csv'))
# co2 = co2[co2['geo'].isin(countryAbb)]
# co2['geo'] = co2['geo'].map(abbrev_to_country).fillna(co2['geo'])
# co2 = co2.pivot_table(columns='TIME_PERIOD', index='geo', values='OBS_VALUE', aggfunc='mean')

# mr = co2.columns[-1] # most recent year
# # co2[[2012, 2016, mr]]/1000 

### 14.4

####   Proportion of fish stocks within biologically sustainable levels

We use two indicators:

1. FMSY/F: catch-weighted average

2. B/BMSY: catch-weighted average


In [19]:
# FMSY/F

In [20]:
# B/BMSY

### 14.5

####  Coverage of protected areas in relation to marine area

We consider two indicators:

1. Coverage of protected areas in relation to marine areas: Distance to Reference transformation where the max-value is set to 30% and the min-value is 0%
2. OHI 'Biodiversity' Index : No further transformation 

In [21]:
# Marine protected areas (% of territorial waters), World Bank aggregation of https://www.protectedplanet.net/en/thematic-areas/marine-protected-areas
# https://data.worldbank.org/indicator/ER.MRN.PTMR.ZS

mpa = pd.read_csv(os.path.join(data_path,'API_ER.MRN.PTMR.ZS_DS2.csv'), skiprows=4)
mpa = mpa[mpa['Country Name'].isin(countries)]
mpa = mpa.set_index('Country Name')
mpa = mpa.dropna(axis=1)
mpa = mpa[['2016', '2021']] # no data for 2012

mpa = (mpa/0.3).round(2)  # dis-ref with target 30%
mpa[mpa>100] = 100
mpa

2016    2021
Country Name                  
Belgium         100.00  100.00
Bulgaria         27.01   27.03
Cyprus            0.41   28.74
Germany         100.00  100.00
Denmark          59.49   61.07
Spain            29.10   42.52
Estonia          62.09   62.60
Finland          37.10   39.96
France           87.22  100.00
United Kingdom   67.45  100.00
Greece            4.86   15.06
Croatia          28.47   29.96
Ireland           7.77    7.77
Italy            29.28   32.45
Lithuania        71.72   85.31
Latvia           53.45   53.46
Malta            20.88   24.82
Netherlands      71.64   89.53
Poland           75.24   75.24
Portugal         27.86   56.06
Romania          28.09   77.00
Sweden           27.35   52.54

In [22]:
# OHI 'Biodiversity' Index 
# https://oceanhealthindex.org/global-scores/data-download/

ohi = pd.read_csv(os.path.join(data_path,'scoresOHI.csv'))
ohi = ohi[ohi['region_name'].isin(countries)]
ohi = ohi[(ohi.long_goal == 'Biodiversity')  & (ohi.dimension == 'score') ]
ohi = ohi.pivot_table( columns='scenario', index='region_name', values='value', aggfunc='mean')

ohi[[2012, 2016, 2022]]


scenario         2012   2016   2022
region_name                        
Belgium         74.65  70.37  71.53
Bulgaria        76.12  67.76  66.19
Croatia         60.77  54.34  55.65
Cyprus          68.48  66.02  64.07
Denmark         72.56  70.68  71.17
Estonia         80.68  75.47  74.96
Finland         79.25  72.17  73.02
France          72.20  69.96  75.08
Germany         73.45  69.15  72.25
Greece          66.28  61.41  59.42
Ireland         67.64  64.11  66.89
Italy           68.72  63.66  64.08
Latvia          77.70  71.71  72.39
Lithuania       76.04  69.74  71.82
Malta           72.23  65.00  64.19
Netherlands     67.14  66.32  68.68
Poland          67.41  64.34  66.45
Portugal        75.73  71.86  72.07
Romania         82.49  71.55  70.55
Spain           72.29  67.51  68.31
Sweden          75.00  73.51  70.96
United Kingdom  68.30  71.82  71.45

### 14.6

####  Degree of implementation of international instruments aiming to combat illegal, unreported and unregulated fishing

We use two indicators:

1. Fishing Subsidies relative to Landings: Max-Min Transformation where the max-value and min-value are the maximum and minimum in the assessment period (2012-2021) 
2. TAC/Catch: 

In [23]:
# Fisheries Support Estimate
# https://stats.oecd.org/Index.aspx?DataSetCode=FISH_FSE

fse = pd.read_csv('./data/FISH_FSE.csv')
fse = fse[fse['Country'].isin(countries)]
# we include  I. DIRECT SUPPORT TO INDIVIDUALS AND COMPANIES and II. SUPPORT FOR SERVICES TO THE SECTOR (except the ones specified below)
fse = fse[fse.Variable.str.startswith(('I.', 'II.')) &\
 ~fse.Variable.isin(('II.E. Education and training', 'II.G. Management of resources', 'II.F. Research and development'))]
# we use US dollars because some countries use their own currency ('Danish Krone', 'Zloty', 'Swedish Krona')
fse= fse[fse.Measure == 'US dollar'][['Country','Variable', 'Value','Year', 'Unit']]

# strip variable codes and delete parent variables to avoid double counting
# solution from https://stackoverflow.com/q/76183612/14534411

separator = '.'
fse['vCode'] = fse.Variable.str.rsplit('.', n=1).str.get(0) 

variableAll = fse.vCode.unique().tolist()

def is_parent(p, target):
    return p.startswith(target) and len(p)>len(target) and p[len(target)] == '.'

vSupport = []
prev = ''
for s in sorted(variableAll)+['']:
    if prev and not is_parent(s, prev):
        vSupport.append(prev)
    prev = s

fse = fse[(fse.vCode.isin(vSupport)) | (fse.vCode.isna())]
fse = fse.groupby(['Country', 'Year']).sum(numeric_only=True).reset_index()

In [24]:
# Even though the USD-EUR discrepancy does not affect the ratio we calculate,
# we get today's exchange rate to convert landing values to USD and have a consistent unit
# Solution source: https://stackoverflow.com/a/17250702/14534411
r = requests.get('http://www.ecb.int/stats/eurofxref/eurofxref-daily.xml', stream=True)
tree = ET.parse(r.raw)
root = tree.getroot()
namespaces = {'ex': 'http://www.ecb.int/vocabulary/2002-08-01/eurofxref'}
currency = 'USD'
match = root.find('.//ex:Cube[@currency="{}"]'.format(currency.upper()), namespaces=namespaces)
eurTOusd = float(match.attrib['rate'])

In [25]:
# Landings of fishery products, TOTAL FISHERY PRODUCTS, Euro, Eurostat
# https://ec.europa.eu/eurostat/databrowser/view/FISH_LD_MAIN/ 
landing = pd.read_csv('./data/fish_ld_main.csv')
landing['Country'] = landing.geo.map(abbrev_to_country).fillna(landing.geo)
landing = landing[landing.Country.isin(countries)]
landing = landing[['Country', 'TIME_PERIOD', 'OBS_VALUE', 'unit']]
landing['landingUSD'] = landing.OBS_VALUE * eurTOusd

In [26]:
# merge subsidies-landings and calculate score
fseLanding = pd.merge(fse, landing, how='left', left_on=['Country', 'Year'], right_on=['Country', 'TIME_PERIOD'])
fseLanding['fseLanding'] = fseLanding.Value / fseLanding.landingUSD 
fseLanding = fseLanding.pivot_table(columns='Year', index='Country', values='fseLanding', aggfunc='mean')
mr = fseLanding.columns[-1] # most recent year
#maxMin transformation as (x - min)/(max-min) * 100 --> converts to 0-100 scale with 0=worst, 100=best
fseScore = (( fseLanding[[2012,2016, mr]] - fseLanding.loc[:,'2012':mr].min().min()) / (fseLanding.loc[:,'2012':mr].max().max()
                                                                   - fseLanding.loc[:,'2012':mr].min().min())*100).round(2)
fseScore

Year             2012   2016   2020
Country                            
Belgium          1.61   0.97    NaN
Denmark          2.10   1.54   2.09
Estonia          8.40   2.14   4.72
France           0.70   0.06    NaN
Germany           NaN   0.07    NaN
Greece           1.73    NaN   0.35
Ireland           NaN   0.80    NaN
Italy            1.50   0.79   1.37
Latvia            NaN   2.48   1.81
Lithuania        4.07  60.28  44.26
Netherlands      0.14   0.20   0.09
Poland          33.42  29.87  82.67
Portugal         0.17   0.20   0.52
Spain            0.71   0.07   0.04
Sweden           7.94   8.74  15.80
United Kingdom   0.21   0.14    NaN

In [27]:
# TAC/Catch

### 14.7

####  Sustainable fisheries as a proportion of GDP in small island developing States, least developed countries and all countries

We use two indicators:

1. OHI Coastal 'Livelihoods & economies' Index: No further transformation
2. OHI 'Tourism & recreation' Index: No further transformation

In [28]:
# OHI 'Livelihoods & economies' Index 
# https://oceanhealthindex.org/global-scores/data-download/

ohi = pd.read_csv(os.path.join(data_path,'scoresOHI.csv'))
ohi = ohi[ohi['region_name'].isin(countries)]
ohi = ohi[(ohi.long_goal == 'Livelihoods & economies')  & (ohi.dimension == 'score')]
ohi = ohi.pivot_table( columns='scenario', index='region_name', values='value', aggfunc='mean')

ohi[[2012, 2016, 2022]]

scenario          2012    2016    2022
region_name                           
Belgium          67.75   80.50   80.50
Bulgaria         71.31   71.31   71.31
Croatia          78.76   78.56   78.56
Cyprus           65.03   63.18   63.18
Denmark          61.44   66.22   66.22
Estonia          59.66   73.81   73.81
Finland          75.44   82.24   82.24
France           79.40   79.41   79.41
Germany          86.50   89.31   89.31
Greece           64.92   61.38   61.38
Ireland          73.29   77.30   77.30
Italy            83.58   85.62   85.62
Latvia           77.74   89.50   89.50
Lithuania        70.18   76.21   76.21
Malta            84.24   86.67   86.67
Netherlands      60.89   62.19   62.19
Poland           80.23   81.50   81.50
Portugal         92.97   93.38   93.38
Romania          90.29   88.77   88.77
Spain            77.74   77.21   77.21
Sweden          100.00  100.00  100.00
United Kingdom   87.45   88.59   88.59

In [29]:
# OHI 'Tourism & recreation' Index 
# https://oceanhealthindex.org/global-scores/data-download/

ohi = pd.read_csv(os.path.join(data_path,'scoresOHI.csv'))
ohi = ohi[ohi['region_name'].isin(countries)]
ohi = ohi[(ohi.long_goal == 'Tourism & recreation')  & (ohi.dimension == 'score')]
ohi = ohi.pivot_table( columns='scenario', index='region_name', values='value', aggfunc='mean')

ohi[[2012, 2016, 2022]]

scenario          2012    2016    2022
region_name                           
Belgium          25.64   24.10   22.89
Bulgaria         23.30   26.04   19.88
Croatia          69.36   88.01   70.06
Cyprus           60.50   63.53   59.75
Denmark          23.47   28.70   22.73
Estonia          38.02   44.89   36.82
Finland          24.10   25.51   19.94
France           44.27   45.31   45.29
Germany          71.18   72.02   61.55
Greece           85.06   98.98  100.00
Ireland          23.52   19.77   21.68
Italy            54.93   65.43   57.67
Latvia           36.83   44.21   30.08
Lithuania        14.03   20.23   14.89
Malta           100.00  100.00  100.00
Netherlands      61.06   59.58   62.31
Poland           16.68   18.12   17.89
Portugal         61.92   79.90   72.35
Romania          20.15   23.35   22.50
Spain            40.22   47.19   41.12
Sweden           37.09   40.77   31.40
United Kingdom   50.05   47.76   50.69

### 14.a

#### Proportion of total research budget allocated to research in the field of marine technology

We use two indicators:

1. Official UNSD indicator ER_RDE_OSEX: Max-Min transformation where the max-value and min-value are the maximum and minimum in the assessment period (2009-2017) of all European countries (not restricted to the countries in the analysis).
2. SAD/TAC: Catch-weighted TAC relative to Scientific Advice 

Note: ER_RDE_OSEX data comes from Global Ocean Science Report (GOSR) 2020, and goes from 2013 to 2017. Data for 2009-2012 data is available in the [UNstats archive](https://unstats.un.org/sdgs/indicators/database/archive) (download csv for 29-Mar-19)

<span style="color:IndianRed"> To discuss: Data for 2009-2012 is only in the archive dataset. The data is quite different, both datasets have year 2013 but the values differ. We need to decide how to account for the difference or to drop the archive.
Wether to use average of the countries with data as replacement for missing values or the World data. Most are missing for the lasts years, maybe we need a different indicator </span>

In [118]:
# %%time
# National ocean science expenditure as a share of total research and development funding (%), UNstats archive (years 2009-2012)
# https://unstats.un.org/sdgs/indicators/database/archive

# # read old data by chunks to speed loading and save 14.a.1 to separate file
# iter_csv = pd.read_csv('./data/AllData_Onward_20190329.csv', iterator=True, chunksize=1000, encoding='cp1252')
# oResearchOld = pd.concat([chunk[chunk.Indicator == '14.a.1'] for chunk in iter_csv])
# oResearchOld.to_csv('./data/archive14a1.csv', index=False)

oResearchOld = pd.read_csv('./data/archive14a1.csv')
oResearchOld = oResearchOld.pivot(index='GeoAreaName', columns='TimePeriod', values='Value')

In [146]:
# National ocean science expenditure as a share of total research and development funding (%), UNstats
# https://unstats.un.org/sdgs/dataportal/database

# read official data and merge with archive data 
oResearch = sdg14[sdg14['SeriesCode']=='ER_RDE_OSEX'].pivot_table(columns='TimePeriod', index='GeoAreaName', values='Value', aggfunc='mean')
oResearch = oResearchOld.merge(oResearch, left_index=True, right_index=True, how='outer')
# use all countries in Europe
oResearch = oResearch[oResearch.index.isin(allEurope)].dropna(how = 'all', axis=1)

mr = oResearch.columns[-1] # most recent year
#maxMin transformation as (x - min)/(max-min) * 100 --> converts to 0-100 scale with 0=worst, 100=best
oResearch = (( oResearch[[2012,2016, mr]] - oResearch.loc[:,2012:mr].min().min()) / (oResearch.loc[:,2012:mr].max().max()
                                                                   - oResearch.loc[:,2012:mr].min().min())*100).round(2)
# filter countries included in the analysis                                                                  
oResearch[oResearch.index.isin(countries)]     
 
# fill NaN with mean of column       
# oResearch.fillna(oResearch.mean())                                                      

TimePeriod    2012   2016  2021
GeoAreaName                    
Belgium       0.17   3.47  1.18
Bulgaria       NaN   0.26   NaN
Croatia        NaN    NaN   NaN
Finland       2.32   4.04   NaN
France         NaN    NaN  7.77
Germany        NaN   3.05  4.30
Ireland        NaN    NaN   NaN
Italy        14.42   3.84   NaN
Netherlands    NaN   2.49   NaN
Poland         NaN   1.16   NaN
Portugal       NaN  33.88   NaN
Romania       7.22    NaN   NaN
Spain         4.61   6.56   NaN

In [33]:
# SAD/TAC
# If a country fishes only on fish stocks where assignment of TAC follows scientific advice, it would score 100

### 14.b

#### Degree of application of a legal/regulatory/policy/institutional framework which recognizes and protects access rights for small‐scale fisheries

We use two indicators: 

1. OHI Artisanal Fishing Opportunities Index: No further transformation
2. Percentage of Fish Species Threatened: No further transformation

In [150]:
# OHI 'Artisanal opportunities' Index
# https://oceanhealthindex.org/global-scores/data-download/

ohi = pd.read_csv(os.path.join(data_path,'scoresOHI.csv'))
ohi = ohi[ohi['region_name'].isin(countries)]
ohi = ohi[(ohi.long_goal == 'Artisanal opportunities')  & (ohi.dimension == 'score')]
ohi = ohi.pivot_table( columns='scenario', index='region_name', values='value', aggfunc='mean')

ohi[[2012, 2017, 2022]]


scenario         2012   2017   2022
region_name                        
Belgium         78.53  77.50  78.78
Bulgaria        55.03  55.40  66.86
Croatia         64.91  62.55  68.79
Cyprus          76.37  70.09  77.83
Denmark         70.31  78.51  73.56
Estonia         89.37  90.97  99.62
Finland         85.84  82.33  74.71
France          76.77  76.16  79.13
Germany         70.41  75.54  71.64
Greece          64.00  60.38  69.62
Ireland         66.92  73.60  73.55
Italy           61.10  64.56  69.66
Latvia          88.51  89.35  99.50
Lithuania       93.90  90.55  98.81
Malta           71.17  72.29  83.15
Netherlands     46.83  56.63  61.19
Poland          91.75  73.12  77.46
Portugal        75.30  59.36  64.94
Romania         67.63  69.50  82.96
Spain           72.27  70.63  73.50
Sweden          96.81  96.15  97.79
United Kingdom  81.84  84.24  78.44

In [ ]:
# Percentage of Fish Species Threatened
# https://fishbase.mnhn.fr/country/CountryChecklist.php?c_code=056&vhabitat=threatened&csub_code=
# But data seems to be for current threatened species, there is no time series
# We can extract the data and calculate the percentage of threatened species
# https://en.wikipedia.org/wiki/ISO_3166-1_numeric 

# https://data.worldbank.org/indicator/EN.FSH.THRD.NO
# Data is only for 2018

#https://www.iucnredlist.org/search


### 14.c
####  Number of countries making progress in ratifying, accepting and implementing through legal, policy and institutional frameworks, ocean-related instruments that implement international law, as reflected in the United Nations Convention on the Law of the Sea

We use two indicators:

1. Participation in agreements of the International Marine Organization (IMO Participation Rate): 
2. Measures under the Marine Strategy Framework Directive: 

In [ ]:
# https://www.imo.org/en/OurWork/ERO/Pages/MemberStates.aspx